Source/WTF/wtf/SHA1.cpp

```diff
@@ -242,13 +242,10 @@ void SHA1::addUTF8Bytes(CFStringRef string)
 
 CString SHA1::hexDigest(const Digest& digest)
 {
-    char* start = nullptr;
+    std::span<char> start;
     CString result = CString::newUninitialized(40, start);
-    char* buffer = start;
-    for (size_t i = 0; i < hashSize; ++i) {
-        snprintf(buffer, 3, "%02X", digest.at(i));
-        buffer += 2;
-    }
+    for (size_t i = 0; i < hashSize; ++i)
+        snprintf(start.subspan(i * 2).data(), 3, "%02X", digest.at(i));
```

Comment:
We have functions in `ASCIICType.h` that turn a byte into hex digits without involving the `snprintf` function; I think they should be used here instead. This code using `snprintf` is probably a bit slow, parsing that format string over and over again, and writes out null byte terminators that are not needed, mostly overwritten by the following hex digits and then the last one just overwrites a null character that is already put there by `CString::newUninitialized`.